### PR TITLE
Add list box

### PR DESCRIPTION
**Pull request by pochretien**

### CODE REVIEW

**Comment by pochretien** [comment]
image

File: src/components/listBox/ListBox.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    },
+  };
+
+  getClasses(): string {
+    return classNames('list-box', this.props.classes);
+  }
+
+  protected getItems(): JSX.Element[] | JSX.Element {
+    const items = _.chain(this.props.items)
+      .map((item: IItemBoxProps) => <ItemBox {..._.extend(item, { onOptionClick: this.props.onOptionClick }) } />)
```
Comment (sroy3) [inline src/components/listBox/ListBox.tsx]: You need a unique key on these items

File: src/components/listBox/ListBox.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+  static defaultProps: Partial<IListBoxProps> = {
+    noResultItem: {
+      value: 'No Items',
+    },
+  };
+
+  getClasses(): string {
+    return classNames('list-box', this.props.classes);
+  }
```
Comment (GermainBergeron) [inline src/components/listBox/ListBox.tsx]: private? No real need for a function here

File: src/components/listBox/ListBox.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+  protected getItems(): JSX.Element[] | JSX.Element {
+    const items = _.chain(this.props.items)
+      .map((item: IItemBoxProps) => <ItemBox {..._.extend(item, { onOptionClick: this.props.onOptionClick }) } />)
+      .value();
+
+    return items.length ? items : <ItemBox {..._.extend(this.props.noResultItem, { classes: ['multi-line'] }) } />;
+  }
+
+  handleOnOptionClick = (option: IItemBoxProps) => {
```
Comment (GermainBergeron) [inline src/components/listBox/ListBox.tsx]: private

File: src/components/listBox/ListBox.tsx [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  getClasses(): string {
+    return classNames('list-box', this.props.classes);
+  }
+
+  protected getItems(): JSX.Element[] | JSX.Element {
+    const items = _.chain(this.props.items)
+      .map((item: IItemBoxProps) => <ItemBox {..._.extend(item, { onOptionClick: this.props.onOptionClick }) } />)
+      .value();
+
+    return items.length ? items : <ItemBox {..._.extend(this.props.noResultItem, { classes: ['multi-line'] }) } />;
```
Comment (GermainBergeron) [inline src/components/listBox/ListBox.tsx]: The line is quite long, can you put the ternary on 3 lines?

File: src/components/listBox/tests/ListBox.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+      listBoxComponent.find(ItemBox).node.handleOnOptionClick({ target: 'target' });
+      expect(onOptionClick).toHaveBeenCalled();
+    });
+
+    it('should show the no result <BoxItem/> if the items array is empty', () => {
+      renderListBox({
+        items: [],
+      });
+
+      expect(listBoxComponent.find(ItemBox).length).toBe(1);
```
Comment (GermainBergeron) [inline src/components/listBox/tests/ListBox.spec.tsx]: The expect is not really coherent with the test describe. Maybe you should check the text or classes instead?

Comment (MichelML) [reply src/components/listBox/tests/ListBox.spec.tsx]: you should also test what is inside the box and check if it contains the string "No items"

Comment (pochretien) [reply src/components/listBox/tests/ListBox.spec.tsx]: test added for the no result box

**Pull request merged**